### ISSUE TITLE
Issue in SugarRecord.operation(...)

### ISSUE THREAD

**Issue by nayzak** [gitdoapp/SugarRecord#135]
I'm using SugarRecord with Realm and just noticed an issue in this code:
```swift
SugarRecord.operation(inBackground: true, stackType:.SugarRecordEngineRealm) { context in
      context.beginWriting()
      // some code
      context.endWriting()
    }
```
This code fails with exception `Realm accessed from incorrect thread`.
In fact `operation` doing this:
```swift
let stack: SugarRecordStackProtocol? = stackFortype(stackType)
let context: SugarRecordContext? = background ? stack!.backgroundContext(): stack!.mainThreadContext()
if background {
  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), { () -> Void in
                closure(context: context!)
  })
}
```
And in `stack!.backgroundContext()` we see this:
```swift
public func backgroundContext() -> SugarRecordContext?
{
  return SugarRecordRLMContext(realmContext: RLMRealm.defaultRealm())
}
```
So, `RLMRealm.defaultRealm()` was called from different thread than `context.beginWriting()`.

**Comment by mheicke** [comment]
same problem with CoreData: the backgroundContext is retrieved before the dispatch. Hence it's highly likable that the context is created on a different thread then it is later used. This leads to several problems (see documentation: https://developer.apple.com/library/prerelease/ios/documentation/Cocoa/Reference/CoreDataFramework/Classes/NSManagedObjectContext_Class/index.html) and random crashes which seem to be solved when retrieving the context inside of the dispatch_async closure

**Comment by pepibumur** [comment]
It'll be fixed in SugarRecord 2.0
